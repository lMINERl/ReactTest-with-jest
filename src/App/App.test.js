import React from 'react';
import ReactDOM from 'react-dom';
import App from './index.js';
import renderer from 'react-test-renderer';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('just a fake test', () => {
  test('fake test', () => {
    expect(true).toBeTruthy();
    expect(false).toBeFalsy();
    // snapshot 
    const component = renderer.create(<div>memory component</div>);
    expect(component).toMatchSnapshot();
  });
});

describe('app component', () => {

  it('doesnt have sections', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains(<section />)).toEqual(false);

    // updating test cases when requirement changes and output changes
    expect(wrapper.contains(<section />)).toMatchSnapshot(`app has section`);
    // serializer
    expect(wrapper.contains(<section />)).toMatchSnapshot(
      false
    );

    // snapshot to dom
    expect(wrapper).toMatchSnapshot();
  });

  it('does have logo', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.exists('.App-logo'));
  });

  it('does show logo', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('img').prop('src') !== null).toBeTruthy();
  })
});


// describe('App Component', () => {
//   it('shows an img', () => {
//     expect(false).toBeFalsy();
//     const app = shallow(App);

//     expect(app.find('img').prop('src')).toEqual(require('../assets/logo.svg'));
//   });
// });