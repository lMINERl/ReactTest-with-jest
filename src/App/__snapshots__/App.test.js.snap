// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component doesnt have sections 1`] = `false`;

exports[`app component doesnt have sections 2`] = `ShallowWrapper {}`;

exports[`app component doesnt have sections: app has section 1`] = `false`;

exports[`just a fake test fake test 1`] = `
<div>
  memory component
</div>
`;
